examples/experimental/backprop-regression/BackpropRegression.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Main where
+
+import Data.Proxy
+import GHC.TypeLits
+
+import Numeric.Backprop as Bp
+import Prelude as P
+import Lens.Micro.TH
+import Torch.Double as Torch hiding (add)
+import Torch.Double.NN.Linear (Linear(..), linear)
+import Torch.Double as Math hiding (Sum, add)
+import qualified Torch.Core.Random as RNG
+
+type NSamples = 2000
+type P = '[1, 2]
+type Precision = Double
+
+seedVal :: RNG.Seed
+seedVal = 3141592653579
+
+data Regression = Regression {
+    _linearLayer :: Linear 2 1
+}
+makeLenses ''Regression
```

Comment:
You (and I, both) should get into the habit of replacing template haskell with [generic-lens](https://hackage.haskell.org/package/generic-lens) which has equal performance, greater flexibility, and no TH (which can nuke compile times if you aren't careful).

Reply:
Haven't used generic-lens, do you have an example?

Reply:
You should be able to just replace the generated lens, `linearLayer`, with `field @"linearLayer"` (the `@` comes from the TypeApplications extension). I _think_ that's all! Since it's a lens, it should also be fully compatible with Justin Le's `^^.` combinator.

Reply:
Okay, uses generic-lens now.